syntax = "proto3";

package admin;

service AdminService {
    rpc GetDashboardStats(GetDashboardStatsRequest) returns (DashboardStatsResponse);
    rpc GetEvents(GetEventsRequest) returns (GetEventsResponse);
    rpc UpdatePolicy(UpdatePolicyRequest) returns (UpdatePolicyResponse);
}

message GetDashboardStatsRequest {}

message DashboardStatsResponse {
    int32 events_today = 1;
    int32 denied_today = 2;
    int32 active_tokens = 3;
    int32 active_dongles = 4;
    int32 active_hosts = 5;
}

message GetEventsRequest {
    int32 limit = 1;
    int32 offset = 2;
    string severity = 3;
    string host = 4;
}

message Event {
    string timestamp = 1;
    string host = 2;
    string event_type = 3;
    string severity = 4;
    string file_path = 5;
    string user_id = 6;
    string result = 7;
    string description = 8;
}

message GetEventsResponse {
    repeated Event events = 1;
    int32 total = 2;
}

message UpdatePolicyRequest {
    string policy_yaml = 1;
}

message UpdatePolicyResponse {
    bool success = 1;
    string error = 2;
    string version = 3;
}
